### DISCUSSION TITLE
ERROR @wdio/runner: Error: Failed to create session ||| webdriverio keep connecting to random ports

### DISCUSSION THREAD

**Discussion by maile0** [webdriverio/webdriverio#12949] [Q&A] [answered]
Hey guys, 
I am trying to setup Appium project with Webdriver IO and Typescript on a corporate windows PC. I understood that, 
it needs to connect to my default project port (4723), but it connected just twice.. then it keeps connecting to random ports and failing to create session. My company is using firewall (Kaspersky). 

FYI: I successfully started my project and running tests on my personal MacOS laptop, where I dont have any firewalls except Apple's default ones. 

![image](https://github.com/webdriverio/webdriverio/assets/103849628/3bc73116-0179-42c1-8660-5b68907daafa)

here is my wdio.conf.ts file: 

![image](https://github.com/webdriverio/webdriverio/assets/103849628/aecda947-604c-4fdd-8af6-ed3b7bfaaab6)
![image](https://github.com/webdriverio/webdriverio/assets/103849628/c08bff22-6ac3-45dc-90fa-1434a0534531)

and my tsconfit.json 

![image](https://github.com/webdriverio/webdriverio/assets/103849628/e8fb4525-933b-4dc5-b3fc-ec5a5780ecc1)

here's my package.json 

![image](https://github.com/webdriverio/webdriverio/assets/103849628/5ba2e39d-0eba-45f9-aca9-e86dfffdeab0)

and my appium server is started on terminal with "appium -p 4724 (the default appium inspector port). 

Please help me,  I am on the verge here :D

**Comment by christian-bromann** [comment, marked as answer]
If you have an Appium server running already, you don't need the Appium service in WDIO as all it does is to start Appium.

**Reply by maile0** [reply]
I see, but I got that error above when I try to run my test. How can I resolve it?

**Reply by christian-bromann** [reply]
Remove the Appium service in your `wdio.conf.ts`

**Reply by maile0** [reply]
Unfortunately still the same error.. it should connect to http://127.0.0.1:4723, but it still trying to connect to random ports..

**Reply by christian-bromann** [reply]
Yes, for that you have to define the `port`, see https://webdriver.io/docs/configuration
